src/samplerate.rs

```diff
@@ -95,8 +95,7 @@ impl Samplerate {
         }
     }
 
-    fn _process(&self, input: &[f32], output_len: usize, end_of_input: bool) -> Result<Vec<f32>, Error> {
-        let channels = self.channels()?;
+    fn _process(&self, input: &[f32], channels: usize, output_len: usize, end_of_input: bool) -> Result<Vec<f32>, Error> {
```

Comment:
Why put channels in here as argument, instead of acccessing it inside the method's body?

Reply:
This is to avoid redundant calls to `self.channels()`. Although I agree this isn't the most elegant solution.

Reply:
If you prefer I can change it back.